lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2262,6 +2262,174 @@ class ArchiveTool : public Tool {
   }
 };
 
+#pragma mark - StaleFileRemovalTool implementation
+
+class StaleFileRemovalCommand : public Command {
+  std::string description;
+
+  std::vector<std::string> expectedOutputs;
+  std::vector<std::string> filesToDelete;
+
+  BuildValue priorValue;
+  bool hasPriorResult = false;
+
+  virtual void configureDescription(const ConfigureContext&, StringRef value) override {
+    description = value;
+  }
+
+  virtual void getShortDescription(SmallVectorImpl<char> &result) override {
+    llvm::raw_svector_ostream(result) << (description.empty() ? "Stale file removal" : description);
+  }
+
+  virtual void getVerboseDescription(SmallVectorImpl<char> &result) override {
+    getShortDescription(result);
+    llvm::raw_svector_ostream(result) << ", stale files: [";
+    for (auto fileToDelete : filesToDelete) {
```

Comment:
We don't usually make the description depend on state which is computed by the build... although it is pretty convenient here. If this works, great, but we may need to adjust this if we ever change the ordering of where we compute the description (and I'm a little worried about races).